goonj/rule_engine.py

```diff
@@ -0,0 +1,31 @@
+
+class RuleEngine(object):
```

Comment:
Why is the rule engine performing business logic of maintaining the count of errors and if threshold has crossed? Shouldn't it be the job of `SmartAlert`.